share/rocm/cmake/ROCMWrapper.cmake

```diff
@@ -0,0 +1,99 @@
+# ######################################################################################################################
+# Copyright (C) 2022 Advanced Micro Devices, Inc.
+# ######################################################################################################################
+
+set(ROCM_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}" CACHE INTERNAL "Path to rocm-cmake files.")
+
+function(rocm_wrap_header_dir DIRECTORY)
+    cmake_parse_arguments(PARSE "" "HEADER_LOCATION" "PATTERNS;GUARDS;WRAPPER_LOCATIONS;OUTPUT_LOCATIONS" ${ARGN})
+    if(NOT PARSE_HEADER_LOCATION)
+        set(PARSE_HEADER_LOCATION "include/${CMAKE_PROJECT_NAME}")
+    endif()
+    if(NOT PARSE_PATTERNS)
+        set(PARSE_PATTERNS "*.h;*.hpp;*.hh;*.hxx;*.inl")
+    endif()
+    foreach(PATTERN IN LISTS PARSE_PATTERNS)
+        list(APPEND QUALIFIED_PATTERNS "${DIRECTORY}/${PATTERN}")
+    endforeach()
+    file (GLOB_RECURSE include_files RELATIVE "${DIRECTORY}" ${QUALIFIED_PATTERNS})
```

Comment:
It should be calculated at configure time, and on each reconfigure.

Reply:
The GLOB is really the only thing I'm not so sure about. That said, I suppose you can just use the filename as the pattern if it's an issue.

Reply:
What is your concern with using the `GLOB` here?

Reply:
The concerns I have with using GLOB are the same pretty much anywhere: (1) it will quietly produce unexpected output if the state of the filesystem is different from what you expect, and (2) it breaks the normal dependency analysis for the CMake files.

That said, this is a temporary, optional feature. It's not worth a lot of effort to make it perfect. Let's just get'er done.